### ISSUE TITLE
Analysis

### ISSUE THREAD

**Issue by DavidBuchanan314** [rabbitscam/rabbitr1#11]
tl;dr: The "leak" seems real, but doesn't prove any of the claims made in the readme.

This statement from Peiyuan Liao, the rabbit CTO, is consistent with what I'm seeing here: https://twitter.com/liaopeiyuan/status/1782922595199033662

So the "leak" is a bit of a nothingburger, containing partial code for the relatively boring process of letting users authenticate with online services through a sandboxed browser session, from which auth tokens etc. can be extracted. You can't infer anything about how LAM does or doesn't work from this.

They likely used "[kiosk escape](https://www.secquest.co.uk/white-papers/introduction-to-kiosk-breakout)" tricks to get code exec *within* the box that runs the browser. Assuming their sandboxing is all set up correctly, this isn't particularly concerning, but it does expose the code that runs *within* the sandbox for analysis. That's what we appear to have here.

The attacker left behind a file named `cdk.log`, which is an artifact of https://github.com/cdk-team/CDK/, a container pentesting tool. They were clearly trying to escape the sandbox and pivot to somewhere more interesting, but I don't think they managed it. I think "part 2" is a bluff, this is all they have (feel free to prove me wrong, lol).

But that doesn't mean there's *nothing* here. Lets look at what we do have.

The most interesting detail to me is a package name list in `repo/typescript/common/base-tsconfig.json`

```
app-caod-download
base-core
base-fastify
base-gcp
base-kubernetes
base-media
base-mongodb
base-nli
base-node
base-playwright
base-serp
cm-browsing-minion-common
cm-bunny-host-common
cm-bunny-host-web-common
cm-preset-bunny-builder-common
cm-preset-bunny-builder-next
cm-preset-bunny-builder-service
cm-quantum-peripheral-common
cm-rabbit-common
cm-rabbit-next
cm-rabbit-service
cm-spotify-client
q-apps
q-apps-common
q-apps-executor
q-auth
q-common
q-conversation-finetuning
q-hole-common
q-hole-next
q-hole-service
q-model
q-proxy
q-q1-common
q-q1-service
q-q1-test-next
q-web-minion-common
q-web-minion-service
q-website
```

The only code *actually* present is for `q-web-minion-*`.

What follows is my speculation based on the names alone:

"q" seems like a codename for the rabbit device (so `q-hole` = rabbit hole).

The problem with trying to log into and interface with consumer-facing services from "the cloud" is that you'll get IP rate limited, blocked as a bot, etc. It would make sense to proxy traffic back out through the user's device, and that's what I'd hope `q-proxy` is about. The big downside with this is that it ~doubles latency and ~halves available bandwidth, magnifying any deficiencies of a flaky 4G connection. This is perhaps partly why their doordash demo chugged so hard. (protip to the team: use a caching proxy, with SSL MitM)

This is a total stab in the dark but my guess is that `bunny-host` is where the LAM action happens, and `bunny-builder` is for LAM training.

`cm-quantum-peripheral-common` might be the wrist-mounted device teased in the launch event.
